### PR TITLE
sys-fs/multipath-tools: Version bump 0.6.4

### PR DESCRIPTION
**Pull request by anthonyryan1**
Gentoo-bug: 604228

* Include patch for sysmacros
* Remove patch for makefile (building still performs a few unnecessary
  operations, but the cost is negligible compared to the cost of
  carrying yet another patch.
* Remove COMMIT_ID
* Remove some unnecessary definitions
* Include a way to disable ceph when building
* Add linux-info to check for CONFIG_DM_MULTIPATH

Package-Manager: Portage-2.3.3, Repoman-2.3.1

### CODE REVIEW

**Comment by anthonyryan1** [comment]
CC @prometheanfire 

Can you let me know why we want to ignore modprobe failures in multipathd.service so I can add a comment to that patch explaining it's necessity and why we want to deviate from the upstream behaviour?

Your original commit is here: https://github.com/gentoo/gentoo/commit/dc818926d59d65070204cc0730b96e2b1feff96d

File: sys-fs/multipath-tools/multipath-tools-0.6.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+IUSE="systemd rbd"
+
+RDEPEND=">=sys-fs/lvm2-2.02.45
+	>=virtual/udev-171
+	dev-libs/libaio
+	dev-libs/userspace-rcu
+	sys-libs/readline:*
```
Comment (SoapGentoo) [inline sys-fs/multipath-tools/multipath-tools-0.6.4.ebuild]: `:0=`

**Comment by prometheanfire** [comment]
@anthonyryan1 yes, it's needed because sometimes stuff is built into the kernel statically, meaning modprobe will fail, even though the device/functions are actually available.

Please readd the sed.

File: sys-fs/multipath-tools/multipath-tools-0.6.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	emake \
+		DESTDIR="${D}" \
+		SYSTEMD=$(get_systemd_pv) \
+		unitdir="$(systemd_get_systemunitdir)" \
+		libudevdir='${prefix}'/"$(get_udevdir)" \
+		install
+
+	newinitd "${FILESDIR}"/rc-multipathd multipathd
+	newinitd "${FILESDIR}"/multipath.rc multipath
+
```
Comment (prometheanfire) [inline sys-fs/multipath-tools/multipath-tools-0.6.4.ebuild]: please readd the sed

    sed -i 's|ExecStartPre=/|ExecStartPre=-/|g' "${D}"/usr/lib/systemd/system/multipathd.service || die

Comment (anthonyryan1) [inline sys-fs/multipath-tools/multipath-tools-0.6.4.ebuild]: The sed has not been removed, someone prior to me converted it to a patch. See multipath-tools-0.6.2-ignore-modprobe-failures.patch

The reasoning there sounds like something we should be discussing upstream.

Comment (prometheanfire) [inline sys-fs/multipath-tools/multipath-tools-0.6.4.ebuild]: Ah, ok.  And yes, it'd be nice to get this merged upstream.

**Comment by Polynomial-C** [comment]
Unfortunately packages maintained by base-system project are not (yet) allowed to be EAPI-6. 
I'm sorry about this as I do not support this decision. Please change the ebuild accordingly.

**Comment by anthonyryan1** [comment]
@Polynomial-C The unstable version in the tree has been EAPI=6 for more than 6 months. Surely you don't expect to downgrade it when it's already been shipped with the newer EAPI?

**Comment by prometheanfire** [comment]
@Polynomial-C  really? when was this policy set?

**Comment by Polynomial-C** [comment]
@prometheanfire dunno exactly. But that's the ruling since when I joined @gentoo/base-system. Never use the latest EAPI so very old systems don't break on upgrades.
Like I said, I don't support that ruling.

**Comment by Polynomial-C** [comment]
@anthonyryan1 well, it's just what I observed in this pull request. I usually don't touch this package as I don't have any use for it. But yeah, downgrading EAPI is not really something that should be done.

**Comment by cardoe** [comment]
Just leave it as EAPI=6. @robbat2 did the initial EAPI=6 and @vapier can come complain at me.

**Comment by robbat2** [comment]
merged

**Pull request closed without merging**